skglm/experimental/pdcd_ws.py

```diff
@@ -0,0 +1,219 @@
+import warnings
+
+import numpy as np
+from numpy.linalg import norm
+
+from numba import njit
+from skglm.utils.jit_compilation import compiled_clone
+from sklearn.exceptions import ConvergenceWarning
+
+
+class PDCD_WS:
+    """Primal-Dual Coordinate Descent solver with working sets.
+
+    Solver inspired by [1] that uses working sets [2].
+
+    Parameters
+    ----------
+    max_iter : int, optional
+        The maximum number of iterations or equivalently the
+        the maximum number solved subproblems.
+
+    max_epochs : int, optional
+        Maximum number of CD epochs on each subproblem.
+
+    p0 : int, optional
+        First working set size.
+
+    tol : float, optional
+        The tolerance for the optimization.
+
+    dual_init : array, shape (n_samples,) default None
+        The initialization of dual variables.
+        If None, they are initialized as the 0 vector ``np.zeros(n_samples)``.
+
+    return_p_objs : bool, default False
+        If True, returns the values of the objective in each iteration.
+        Otherwise returns an empty array.
+
+    verbose : bool or int, default False
+        Amount of verbosity. 0/False is silent.
+
+    References
+    ----------
+    .. [1] Olivier Fercoq and Pascal Bianchi,
+        "A Coordinate-Descent Primal-Dual Algorithm with Large Step Size and Possibly
+        Nonseparable Functions", SIAM Journal on Optimization, 2020,
+        https://epubs.siam.org/doi/10.1137/18M1168480,
+        code: https://github.com/Badr-MOUFAD/Fercoq-Bianchi-solver
+
+    .. [2] Mathurin Massias, Alexandre Gramfort, Joseph Salmon,
```

Comment:
this one is too short and does not contain many details

I'd cite the skglm paper instead